Dockerfile

```
@@ -491,45 +453,115 @@ ENV CGO_CFLAGS="${CFLAGS}"
 ENV CGO_CXXFLAGS="${CXXFLAGS}"
 ENV CGO_LDFLAGS="${LDFLAGS}"
 
+ENV GO111MODULE="auto"
+
+ENV AWS_LC_FIPS_VER="2.0.9"
+
+USER root
+RUN dnf -y install golang
+
+ENV GO121VER="1.21.9"
+ENV GO122VER="1.22.1"
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-prep as sdk-go-1.21-prep
+
```

Consider hoisting the "1.21" specifics (version, release) into generic variables at the start of the stage.

```suggestion
FROM sdk-go-prep as sdk-go-1.21-prep

ENV GOVER="${GO121VER}" # relies 
ENV GOREL="${GO121REL}" # can set this to "1.21"
```

The advantage is that then you can refer to `GOVER` and `GOREL` through the stage, which can then be freely copy/pasted into new stages and you just have to tweak these two variables. This should make it easier to add new builds of Go in the future since only a couple lines will be different per stage.

---

`GOREL` isn't a very good variable name, maybe `GOMAJOR`?

---

Sure, I'll take this suggestion. I agree it will make adding new builds of Go in the future a bit easier, but we'll have to think about how `update-go.sh` will account for this (just spitballing, it could grep for a major/minor version, find the version's "prep" step, and update `GOVER` to the appropriate major.minor.patch)